### ISSUE TITLE
Support for official API?

### ISSUE THREAD

**Issue by wasabigeek** [danmurphy1217/notion-ruby#39]
Hi @danmurphy1217 - thanks for this gem! Was wondering if you're planning to add support for the official API? Would love to help if so.

**Comment by danmurphy1217** [comment]
Hi @wasabigeek, yes I was thinking of re-designing this gem to support the official notion API. I'm busy right now (graduating from college, etc.) but can get started on it within the next 10 days.

**Comment by danmurphy1217** [comment]
I don't think that re-factoring this will be too tough- as a matter of fact it will make things much easier (no need to build long JSON hashes). My time is a little tight but I would love to get a small team of people together to split up the work on this.

**Comment by wasabigeek** [comment]
Might be able to draw inspiration for the classes from Notion's official JS types: https://github.com/makenotion/notion-sdk-js/blob/main/src/api-types.ts. Happy to chip in where possible 👍

**Comment by lucasmenezesds** [comment]
Hello @danmurphy1217, I'm interested on helping you with it, how could help you? :)

**Comment by danmurphy1217** [comment]
@wasabigeek yeah great point- I want to keep this OOP-focused so building out specific classes for blocks is the end goal. I will set up a `spec.md` file this weekend that outlines needed classes, folder structure, etc., and then y'all are welcome to attack things from whichever angle you see fit. I will also start working on it early next week. Thanks so much for your support, it really is neat to have people using a little package that was initially built for fun😄 


cc: @lucasmenezesds

**Comment by danmurphy1217** [comment]
@wasabigeek @lucasmenezesds v1 spec for blocks (not including databases and some of the more complicated things) is outlined [here](https://github.com/danmurphy1217/notion-ruby/blob/master/notion_api_block_spec.yaml). I'm going to create a `v2` folder that we can use to rebuild this API
